File: server/app/core/firebase.py
# app/core/firebase.py

import firebase_admin
from firebase_admin import credentials, firestore, storage
import json

# Carga las credenciales desde el archivo JSON
with open("firebase_credentials.json") as f:
    cred_dict = json.load(f)

cred = credentials.Certificate(cred_dict)

# Usa el storageBucket directamente del JSON
bucket_name = cred_dict.get("storageBucket")

# Inicializa la App e incluye el nombre de tu bucket de Cloud Storage
firebase_admin.initialize_app(cred, {
    "storageBucket": bucket_name
})

# Cliente de Firestore
db = firestore.client()

# Bucket de Storage (ahora ya conoce el bucket por defecto)
bucket = storage.bucket()

File: project/src/firebase.ts
// src/firebase.ts
import { initializeApp } from 'firebase/app';
import { getStorage }    from 'firebase/storage';

// Tu JSON con el resto de las credenciales
import firebaseConfigJSON from './firebase_credentials.json';

// Extiende el JSON para añadir el storageBucket
const firebaseConfig = {
  ...firebaseConfigJSON
};

// Inicializa la app con el config completo
const app = initializeApp(firebaseConfig);

// Exporta storage y, si la necesitas, la app
export const storage = getStorage(app);
export { app };
